### PR TITLE
Feature/hello world

### PR DESCRIPTION
**Pull request by kensayo**
- Create backend with RoR
- API response with random message
- Front end created using React
- Redux for state management

### CODE REVIEW

File: app/controllers/v1/messages_controller.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module V1
+  class MessagesController < ApplicationController
+    def index
+      message = Message.order(Arel.sql('RANDOM()')).first.message
+      # message = Message.all.pluck(:message)
```
Comment (rukundoeric) [inline app/controllers/v1/messages_controller.rb]: - [ ] <b>REQUIRED:</b> To make sure that your codebase is always clean, remember to remove commented-codes.

File: app/javascript/components/Greeting.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import React from "react";
+import { connect } from 'react-redux';
+import { createStructuredSelector } from "reselect";
+
+const GET_MESSAGES_REQUEST = 'GET_MESSAGES_REQUEST';
+const GET_MESSAGES_SUCCESS = 'GET_MESSAGES_SUCCESS';
+
+function getMessages() {
+  console.log("Action!");
```
Comment (rukundoeric) [inline app/javascript/components/Greeting.js]: - [ ] <b>REQUIRED:</b> Kindly keep your codebase clean by making sure to remove all debugging code like `console.log` when you no longer need them.

File: app/javascript/components/Greeting.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  return async dispatch => {
+    dispatch({ type: GET_MESSAGES_REQUEST });
+    try {
+      const response = await fetch(`http://localhost:3000/v1/message`);
+      const json = await response.json();
+      console.log("JSON: " + json);
```
Comment (rukundoeric) [inline app/javascript/components/Greeting.js]: - [ ] <b>REQUIRED:</b> Kindly keep your codebase clean by making sure to remove all debugging code like `console.log` when you no longer need them.

File: app/javascript/components/Greeting.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    type: GET_MESSAGES_SUCCESS,
+    json
+  }
+}
+
+class Greeting extends React.Component{
+  render() {
+
+    const { messages } = this.props;
+    console.log(messages);
```
Comment (rukundoeric) [inline app/javascript/components/Greeting.js]: - [ ] <b>REQUIRED:</b> Kindly keep your codebase clean by making sure to remove all debugging code like `console.log` when you no longer need them.

File: app/javascript/components/configureStore.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import { applyMiddleware, createStore } from 'redux';
+import thunk from 'redux-thunk';
+
+const GET_MESSAGES_SUCCESS = 'GET_MESSAGES_SUCCESS'
+const initialState = {
+  messages: []
+}
+
+function rootReducer(state, action) {
+  console.log(action.type);
```
Comment (rukundoeric) [inline app/javascript/components/configureStore.js]: - [ ] <b>REQUIRED:</b> Kindly keep your codebase clean by making sure to remove all debugging code like `console.log` when you no longer need them.

**Review by rukundoeric — CHANGES_REQUESTED** [review]
Hi @kensayo, 

_Great job so far 👏 💯 👍. However, there are few issues we need to address before we move on. Please, check the comments below._
## Highlights
  - Follows git workflow.  💯
  - Codebase looks clean and professional. 💯 

## Required Changes ♻️
  - [ ] Check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
  - [ ] Check the comments under the review.

Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
> As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by addod19 — APPROVED** [review]
### Hi @kensayo, 

Your project is complete!  Congratulations for keeping the codebase clean :partying_face: There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**